SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -186,7 +186,9 @@ - (void)run
 
 - (void)didSucceed
 {
-    OWSAssert(self.message.messageState == TSOutgoingMessageStateSent);
+    if (self.message.messageState != TSOutgoingMessageStateSent) {
+        OWSProdLogAndFail(@"%@ unexpected message status: %@", self.logTag, self.message.statusDescription);
```

Comment:
There's a bug here that I haven't fixed yet. This logging should make fixing it a lot easer.